torchvision/prototype/transforms/_color.py

```diff
@@ -55,74 +54,52 @@ def _check_input(
 
     def _image_transform(
         self,
-        input: T,
+        inpt: T,
         *,
         kernel_tensor: Callable[..., torch.Tensor],
         kernel_pil: Callable[..., PIL.Image.Image],
         **kwargs: Any,
     ) -> T:
-        if isinstance(input, features.Image):
-            output = kernel_tensor(input, **kwargs)
-            return features.Image.new_like(input, output)
-        elif is_simple_tensor(input):
-            return kernel_tensor(input, **kwargs)
-        elif isinstance(input, PIL.Image.Image):
-            return kernel_pil(input, **kwargs)  # type: ignore[no-any-return]
+        if isinstance(inpt, features.Image):
```

Comment:
Nit: this looks like a remnant method that is not longer used from the time we didn't have a middle layer. I think it should be removed and we need to do a quick review and clean up of any similar logic from the codebase after we merge this.

Reply:
Thanks for catching that !